src/components/Documents/DocumentCard.jsx

```diff
@@ -0,0 +1,233 @@
+// React Imports
+import React, { useState } from 'react';
+// Material UI Imports
+import Box from '@mui/material/Box';
+import Button from '@mui/material/Button';
+import Card from '@mui/material/Card';
+import CardContent from '@mui/material/CardContent';
+import FileOpenIcon from '@mui/icons-material/FileOpen';
+import ShareIcon from '@mui/icons-material/Share';
+import DeleteOutlineOutlinedIcon from '@mui/icons-material/DeleteOutlineOutlined';
+import IconButton from '@mui/material/IconButton';
+import Menu from '@mui/material/Menu';
+import MenuItem from '@mui/material/MenuItem';
+import MoreVertIcon from '@mui/icons-material/MoreVert';
+import Typography from '@mui/material/Typography';
+import { useTheme } from '@mui/material/styles';
+// Utility Imports
+import { truncateText, getTypeText } from '@utils';
+
+/**
+ * @typedef {object} Document
+ * @property {string} id - The unique id of the document
+ * @property {string} name - The given name of the document
+ * @property {string} type - The given type of the document
+ * @property {string} description - The given description of the document
+ * @property {string} uploadDate- The upload date of the document
+ * @property {string} endDate - The expiration date of the document
+ * @property {string} fileUrl - The file URL of the document
+ */
+
+/**
+ * DocumentCard - A component that displays a single document's information
+ * and provides actions (preview, share, delete) through a context menu.
+ *
+ * @memberof Documents
+ * @name DocumentCard
+ * @param {object} props - Component props
+ * @param {Document} props.document - The document object to be displayed
+ * @param {Function} props.onPreview - Function to handle the "Preview" action
+ * @param {Function} props.onShare - Function to handle the "Share" action
+ * @param {Function} props.onDelete - Function to handle the "Delete" action
+ * @returns {React.JSX.Element} The DocumentCard component
+ */
+const DocumentCard = ({ document, onShare, onDelete, onPreview }) => {
+  const theme = useTheme();
+
+  const [anchorEl, setAnchorEl] = useState(null);
+  const [openMenu, setOpenMenu] = useState(null);
+
+  // Opens the context menu
+  const handleClick = (event, clickedDocument) => {
+    setAnchorEl(event.currentTarget);
+    setOpenMenu(clickedDocument.id);
+  };
+
+  // Closes the context menu
+  const handleClose = () => {
+    setAnchorEl(null);
+    setOpenMenu(null);
+  };
+
+  // Handles the click on a context menu item, executing the corresponding action
+  const handleMenuItemClick = (action, clickedDocument) => () => {
+    action(clickedDocument);
+    handleClose();
+  };
+
+  // ... (Styling definitions below)
+
+  const rowStyling = {
+    display: 'flex',
+    columnGap: '10px'
+  };
+
+  const columnStyling = {
+    display: 'flex',
+    flexDirection: 'column'
+  };
+
+  const dataStyling = {
+    ...rowStyling,
+    color: 'text.secondary',
+    fontSize: '.875rem',
```

Comment:
Is there a particular reason `rem` is used here when `px` is used almost every other time?

Reply:
Not really, I think at the time it was easier for me to scale font sizes to each other that way. I can change it to px for consistency.